app/controllers/downloads_controller.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+class DownloadsController < ApplicationController
+  include Hydra::Controller::DownloadBehavior
+
+  def show
+    send_content
+  end
+
+  class FileWithMetadata < Dry::Struct
+    delegate :size, :read, :stream, to: :file
+    attribute :id, Valkyrie::Types::Any
+    attribute :file, Valkyrie::Types::Any
+    attribute :mime_type, Valkyrie::Types::Any
+    attribute :original_name, Valkyrie::Types::Any
+  end
+
+  def load_file
+    binary_file = storage_adapter.find_by(id: file_identifier)
+    FileWithMetadata.new(id: file_identifier, file: binary_file, mime_type: asset.mime_type, original_name: asset.original_filename)
+  end
+
+  # Customize the :download ability in your Ability class, or override this method
+  def authorize_download!
+    authorize! :download, asset
+  end
+
+  def prepare_file_headers
+    send_file_headers! content_options
```

Comment:
This whole method is copied/pasted with the only line being changed being setting the modified_date, so I could extract it to a method until #68 is closed. I'll add a comment.